tools/myst/host/strace.c

```diff
@@ -144,6 +145,56 @@ int myst_strace_parse_config(
         ret = myst_set_strace_filter(num_tokens, tokens, strace_config, 1);
     }
 
+    if (cli_getopt(argc, argv, "--strace-filter-tid", &tid_filter) == 0 &&
+        tid_filter)
+    {
+        if (myst_strsplit(tid_filter, ":", &tokens, &num_tokens) != 0)
+        {
+            fprintf(
+                stderr,
+                "Invalid strace-filter-tid option '%s' specified.\n",
+                tid_filter);
+            abort();
+        }
+
+        for (size_t i = 0; i < num_tokens; ++i)
+        {
+            long tid;
+            const char* _tid = tokens[i];
+
+            ret = myst_str2long(_tid, &tid);
+
+            if (tid <= 0)
+            {
+                fprintf(
+                    stderr,
+                    "Tid %s is invalid. Must be a positive number.\n",
+                    _tid);
+                abort();
+            }
+
+            strace_config->tid_trace[i] = tid;
+
+            if (i >= MYST_MAX_IDS)
```

Comment:
Should the `if` clause comes before `strace_config->tid_trace[i] = tid;`?

Reply:
Or give warning before entering the for loop

Reply:
Added this!